src/maker.rs

```diff
@@ -141,45 +173,55 @@ impl Maker {
+                                // We are buying BTC for DAI
+                                // Given an order rate of: 1:9000
+                                // It is NOT profitable to buy, if the current rate is greater than the order rate.
+                                // 1:8800 -> We give less DAI for getting BTC -> Good.
+                                // 1:9200 -> We have to give more DAI for getting BTC -> Sucks.
+                                if order_rate > current_profitable_rate {
+                                    return Ok(TakeRequestDecision::RateNotProfitable);
+                                }
+
+                                let updated_dai_reserved_funds =
+                                    self.dai_reserved_funds.clone() + order.quote;
+                                if updated_dai_reserved_funds > dai_balance {
+                                    return Ok(TakeRequestDecision::InsufficientFunds);
+                                }
+
+                                self.dai_reserved_funds = updated_dai_reserved_funds;
+                            }
+                            None => anyhow::bail!(BalanceNotAvailable(dai::SYMBOL.to_owned())),
                         }
-
-                        self.dai_reserved_funds = updated_dai_reserved_funds;
                     }
                     order
                     @
                     BtcDaiOrder {
                         position: Position::Sell,
                         ..
                     } => {
-                        // We are selling BTC for DAI
-                        // Given an order rate of: 1:9000
-                        // It is NOT profitable to sell, if the current rate is smaller than the order rate.
-                        // 1:8800 -> We get less DAI for our BTC -> Sucks.
-                        // 1:9200 -> We get more DAI for our BTC -> Good.
-                        if order_rate < current_profitable_rate {
-                            return Ok(TakeRequestDecision::RateNotProfitable);
-                        }
-
-                        let updated_btc_reserved_funds =
-                            self.btc_reserved_funds + order.base + self.btc_fee;
-                        if updated_btc_reserved_funds > self.btc_balance {
-                            return Ok(TakeRequestDecision::InsufficientFunds);
+                        match self.btc_balance {
+                            Some(btc_balance) => {
+                                // We are selling BTC for DAI
+                                // Given an order rate of: 1:9000
+                                // It is NOT profitable to sell, if the current rate is smaller than the order rate.
+                                // 1:8800 -> We get less DAI for our BTC -> Sucks.
+                                // 1:9200 -> We get more DAI for our BTC -> Good.
+                                if order_rate < current_profitable_rate {
```

Comment:
:upside_down_face: We could write an `is_as_good_as(rate)` method on `BtcDaiOrder` which hides this complexity and makes it more readable.

Reply:
Soooo much love - I implemented it 😄